### PR TITLE
ILS-166 Egress for ibm-license-service-reporter

### PR DESCRIPTION
**Pull request by karol-wozniak-ibm**
What this PR does / why we need it:
to solve https://jsw.ibm.com/browse/ILS-166

Which issue(s) this PR fixes:
Fixes # https://jsw.ibm.com/browse/ILS-246

Special notes for your reviewer:
[ILS-246-Tests.pdf](https://github.com/user-attachments/files/17750848/ILS-246-Tests.pdf)

How to backport this PR to other branch:

Add label to this PR with the target branch name backport <branch-name>
The PR will be automatically created in the target branch after merging this PR
If this PR is already merged, you can still add the label with the target branch name backport <branch-name> and leave a comment /backport to trigger the backport action

### CODE REVIEW

**Review by pgodowski — CHANGES_REQUESTED** [review]
pls apply the same review comments as in https://github.com/IBM/ibm-common-service-operator/pull/2303

File: cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-egress.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: ibm-license-service-reporter-deny-all-egress
```
Comment (pgodowski) [inline cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-egress.yaml:4]: please remove this file from the PR - thanks.

File: cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml [hunk 2]
```diff
@@ -1,17 +1,15 @@
 apiVersion: networking.k8s.io/v1
```
Comment (pgodowski) [inline cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml:1]: please do not rename the filename - keep the original filename

File: cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml [hunk 3]
```diff
@@ -1,17 +1,15 @@
...
 kind: NetworkPolicy
 metadata:
-  name: egress-ibm-license-service-reporter-operator
-  namespace: "lsrNamespace"
-  labels:
-    component: cpfs3
+  name: ibm-license-service-reporter-operator-kubernetes-egress
+  namespace: ibm-licensing-namespace
```
Comment (pgodowski) [inline cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml:5]:
```suggestion
  namespace: lsrNamespace
```

Comment (pgodowski) [reply cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml:5]: please use the templated namespace name - thanks.

**Review by pgodowski — CHANGES_REQUESTED** [review]

File: cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml (hunk 2 above)
Comment (karol-wozniak-ibm) [reply cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml]: file name doesn't matter. 
.sh running everything from folder
will leave "ibm-license-service-reporter-operator-egress.yaml" - more readable for what and what is in the content

File: cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml (hunk 3 above)
Comment (karol-wozniak-ibm) [reply cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml]: done

File: cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml [hunk 4]
```diff
@@ -3,15 +3,13 @@ kind: NetworkPolicy
 metadata:
   name: egress-ibm-license-service-reporter-operator
   namespace: "lsrNamespace"
-  labels:
```
Comment (pgodowski) [inline cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml:6]: pls restore the labels.
please DO NOT rename filename.

**Review by pgodowski — CHANGES_REQUESTED** [review]

File: cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml (hunk 2 above)
Comment (pgodowski) [reply cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml]: I don't agree - for the upgrade scenario, the old networkpolicy will not be updated

File: cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml (hunk 4 above)
Comment (karol-wozniak-ibm) [reply cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml]: done

File: cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml (hunk 2 above)
Comment (karol-wozniak-ibm) [reply cp3-networkpolicy/egress/license-service-reporter/ibm-license-service-reporter-operator-egress.yaml]: changed in other comment

**Review by pgodowski — APPROVED** [review]

**Comment by pgodowski** [comment]
@bitscuit Can you pls review/merge? I'm fine with the updates as they are now. Thanks.

**Review by bitscuit — APPROVED** [review]
/lgtm

**Pull request merged**